src/backend/distributed/executor/local_executor.c

```diff
@@ -585,6 +588,39 @@ ExecuteLocalTaskPlan(PlannedStmt *taskPlan, char *queryString,
 }
 
 
+/*
+ * RecordReferenceTableAccessesForTask records relation accesses for reference
+ * tables that given task will access.
+ */
+static void
+RecordReferenceTableAccessesForTask(Task *task)
+{
+	ShardPlacement *taskPlacement = NULL;
+	foreach_ptr(taskPlacement, task->taskPlacementList)
```

Comment:
As @marcocitus noted in #4133, we don't need to iterate over `placementList`, we  should do it only for the first task placement, we are interested in the relations and tasks placements point to the same relations